plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/commands/SQLCommandInclude.java

```diff
@@ -166,30 +193,19 @@ public void onEndQuery(DBCSession session, SQLQueryResult result, DBCStatistics
 
         @Override
         public void onEndScript(DBCStatistics statistics, boolean hasErrors) {
-            if (editor.getActivePreferenceStore().getBoolean(SQLPreferenceConstants.CLOSE_INCLUDED_SCRIPT_AFTER_EXECUTION)) {
+            if (isShouldCloseIncludedScript(hasErrors)) {
                 UIUtils.syncExec(() -> workbenchWindow.getActivePage().closeEditor(editor, false));
             }
-            statusFlag[0] = true;
         }
 
         @Override
         public void onEndSqlJob(DBCSession session, SqlJobResult result) {
-
+            this.result.complete(result.equals(SqlJobResult.SUCCESS) ? SQLControlResult.success() : SQLControlResult.failure());
```

Comment:
Compare enums with `==`

```suggestion
            this.result.complete(result == SqlJobResult.SUCCESS ? SQLControlResult.success() : SQLControlResult.failure());
```